src/plugins/ui_feature/find_and_replace.ts

```diff
@@ -216,45 +329,57 @@ export class FindAndReplacePlugin extends UIPlugin {
   /**
    * Replace the value of the currently selected match
    */
-  private replace(replaceWith: string) {
-    if (this.selectedMatchIndex === null || !this.currentSearchRegex) {
+  private replace(replaceWith: string, updateSearch: boolean = true) {
+    const selectedMatch =
+      this.selectedMatchIndex !== null ? this.searchMatches[this.selectedMatchIndex] : null;
```

Comment:
Opinion : early return if this.selectedMatchIndex is null is more redable than a ternary